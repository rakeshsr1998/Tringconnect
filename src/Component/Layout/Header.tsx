import React from 'react';
import {Image, Pressable, View} from 'react-native';
import {style} from './header-style';

export const Header = () => {
  const [menu, setMenu] = React.useState(false);
  const [search, setSearch] = React.useState(false);
  const [bell, setBell] = React.useState(false);
  const [showDot, setShowDot] = React.useState(true);

  return (
    <View style={style.container}>
      <View style={style.subContainer}>
        <Pressable onPress={() => setMenu(!menu)}>
          <Image
            source={
              menu
                ? require('../../assets/active-menu.png')
                : require('../../assets/menu.png')
            }
            style={style.menu}
          />
        </Pressable>
        <View>
          <Image source={require('../../assets/logo.png')} style={style.logo} />
        </View>
        <View style={style.row}>
          <Pressable
            onPress={() => {
              setBell(!bell);
              setShowDot(false);
            }}>
            {showDot && <View style={style.dot} />}
            <Image
              source={
                bell
                  ? require('../../assets/active-bell.png')
                  : require('../../assets/bell.png')
              }
              style={style.menu}
            />
          </Pressable>
          <Pressable onPress={() => setSearch(!search)}>
            <Image
              source={
                search
                  ? require('../../assets/active-search.png')
                  : require('../../assets/search.png')
              }
              style={style.bell}
            />
          </Pressable>
        </View>
      </View>
    </View>
  );
};
