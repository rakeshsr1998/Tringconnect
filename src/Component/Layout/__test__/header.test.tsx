import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {Header} from '../Header';

describe('<Header>', () => {
  const mockFunction = jest.fn();
  beforeEach(() => {
    const component = <Header />;
    instance = render(component);
  });

  /**
   * @type {RenderAPI}
   */
  let instance: any;

  it('Should render header with menu click', async () => {
    const nextButton = instance.getByTestId('header');
    fireEvent(nextButton, 'onPress');
    expect(mockFunction).toMatchSnapshot();
  });

  it('Should render header with bell click', async () => {
    const nextButton = instance.getByTestId('bell');
    fireEvent(nextButton, 'onPress');
    expect(mockFunction).toMatchSnapshot();
  });

  it('Should render header with notification click', async () => {
    const nextButton = instance.getByTestId('notification');
    fireEvent(nextButton, 'onPress');
    expect(mockFunction).toMatchSnapshot();
  });
});
