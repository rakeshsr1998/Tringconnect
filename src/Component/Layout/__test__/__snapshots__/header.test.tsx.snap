// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> Should render header with bell click 1`] = `[MockFunction]`;

exports[`<Header> Should render header with click 1`] = `[MockFunction]`;

exports[`<Header> Should render header with menu click 1`] = `[MockFunction]`;

exports[`<Header> Should render header with notification click 1`] = `[MockFunction]`;
